src/native/thrift/nativeConnection.ts

```diff
@@ -123,5 +126,12 @@ export function createAppClient<TClient extends ThriftClient<void>>(
     transport: TransportType = 'buffered', 
     protocol: ProtocolType = 'compact'
 ): TClient {
-    return new ServiceClient(new NativeConnection(getTransport(transport), getProtocol(protocol)));
+    class NamedMultiplexedProtocol extends TMultiplexedProtocol {
```

Comment:
If it's now multuplexed, should `createAppClient ` be renamed? I'm assuming we'll be making two, one for `webview.thrift` and for `native.thrift`?

Reply:
We won't create one for `webview.thrift` and `native.thrift`.  We might split `native.thrift` out into
```
service Native {
    i32 nativeThriftPackageVersion()
}

service Commercial {
    void insertAdverts(1:list<AdSlot> adSlots),
    void updateAdverts(1:list<AdSlot> adSlots)
}

service Acquistions {
    void launchFrictionScreen(),
    MaybeEpic getEpics(),
    void epicSeen()
}

service Notifications {
    void follow(1:Topic topic),
    void unfollow(1:Topic topic),
    bool isFollowing(1:Topic topic),
}

service User {
    bool isPremium()
}

service ??? {
    void launchSlideshow(1:list<Image> images, 2:i32 selectedIndex)
}
```

All of these would be app clients, that could be created by, for example:
`createAppClient<Acquistions.Client>(Acquistions.Client, 'buffered', 'compact');`

Although maybe it would make sense to just be `createClient`?

Reply:
I see, that will be useful as the number of APIs grow. `createAppClient` sounds ok knowing that, as it shows it's calling into native code (although that might not be to everyone)

Reply:
I was reading `multiplex` as being able to call into native and having native use the same code to call into the webview

Reply:
Ah, I see there is scope for confusion there.